src/RegisterMapping.cpp

```diff
@@ -385,11 +347,25 @@ bool RegisterMapping::update(const std::vector<uint16_t>& newValues)
     bool isValid = m_isValid;
     m_isValid = true;
 
+    m_lastUpdateTime = std::chrono::high_resolution_clock::now();
+
     return !isValueInitialized || different || !isValid;
 }
 
 bool RegisterMapping::doesUpdate(bool newRegisterValue) const
 {
+    if (!m_isInitialized)
```

Comment:
As above, do not ignore m_isValid. It is used to report value when the previous read has failed

Reply:
Still missing the m_isValid check as it will not reach the last return statement if rejected by freq filter